File: T4/codigos2122/Listas_Modelos/app/src/main/java/com/example/listas/ListaActivity.java
package com.example.listas;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;

import com.example.listas.adapters.AdaptadorLista;
import com.example.listas.utils.Modelo;

import java.util.ArrayList;

public class ListaActivity extends AppCompatActivity {

    private ListView lista;
    private ArrayList<Modelo> listaModelos;
    private AdaptadorLista adaptadorLista;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_lista);
        instancias();
        rellenarListas();
        asociarElementos();
    }

    private void rellenarListas() {
        listaModelos.add(new Modelo("I8","BMW",R.drawable.i8,300));
        listaModelos.add(new Modelo("EQC","Mercedes",R.drawable.eqc,400));
        listaModelos.add(new Modelo("EQC","Mercedes",R.drawable.eqc,400));
        listaModelos.add(new Modelo("EQC","Mercedes",R.drawable.eqc,400));
        listaModelos.add(new Modelo("EQC","Mercedes",R.drawable.eqc,400));
        listaModelos.add(new Modelo("EQC","Mercedes",R.drawable.eqc,400));
        listaModelos.add(new Modelo("A5","Audi",R.drawable.a5,500));
        listaModelos.add(new Modelo("A5","Audi",R.drawable.a5,500));
        listaModelos.add(new Modelo("A5","Audi",R.drawable.a5,500));
        listaModelos.add(new Modelo("A5","Audi",R.drawable.a5,500));
        listaModelos.add(new Modelo("A5","Audi",R.drawable.a5,500));
        listaModelos.add(new Modelo("Arteon","VW",R.drawable.arteon,200));
        listaModelos.add(new Modelo("Arteon","VW",R.drawable.arteon,200));
        listaModelos.add(new Modelo("Arteon","VW",R.drawable.arteon,200));
        listaModelos.add(new Modelo("Arteon","VW",R.drawable.arteon,200));
        listaModelos.add(new Modelo("Arteon","VW",R.drawable.arteon,200));
        adaptadorLista.notifyDataSetChanged();
    }

    private void asociarElementos() {
        lista.setAdapter(adaptadorLista);
    }

    private void instancias() {
        lista = findViewById(R.id.lista);
        listaModelos = new ArrayList<>();
        adaptadorLista = new AdaptadorLista(listaModelos,getApplicationContext());
    }
}

File: Kotlin/T3/codigos_ces/FiltradoRestaurantes/app/src/main/java/com/example/filtradorestaurantes/ListaActivity.kt
package com.example.filtradorestaurantes

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.filtradorestaurantes.data.DataSet
import com.example.filtradorestaurantes.model.Restaurante

class ListaActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lista)

        // obtener la lista filtrada
        val localidad = intent.extras!!.getString("localidad")
        val comida = intent.extras!!.getString("comida")
        val puntuacion = intent.extras!!.getInt("puntuacion")

        val listaFiltrada: ArrayList<Restaurante>
        = DataSet.listaRestaurantes().filter {
            it.comida.equals(comida)
                    && it.puntuacion >= puntuacion
                    && it.localidad.equals(localidad)
        } as ArrayList<Restaurante>

    }
}